src/EasyAbp.BookingService.Domain/EasyAbp/BookingService/AssetOccupancyProviders/AssetOccupancyProviderBase.cs

```diff
@@ -264,66 +242,86 @@ public virtual async Task<CanOccupyResult> CanOccupyByCategoryAsync(OccupyAssetB
         return CanBulkOccupyResult.Success;
     }
 
-    protected virtual async Task<List<PeriodOccupancyModel>> GetCachedAssetDayPeriodsAsync(Asset asset,
-        AssetCategory category,
-        DateTime date,
-        Dictionary<(Guid, DateTime), List<PeriodOccupancyModel>> cachedPeriods,
-        PeriodScheme periodScheme = default)
-    {
-        if (!cachedPeriods.ContainsKey((asset.Id, date)))
-        {
-            cachedPeriods[(asset.Id, date)] =
-                periodScheme is null
-                    ? await GetPeriodsAsync(asset, category, date)
-                    : await InternalGetPeriodsAsync(asset, category, periodScheme, date);
-        }
-
-        return cachedPeriods[(asset.Id, date)];
-    }
-
     [UnitOfWork(true)]
     public virtual async Task<(ProviderAssetOccupancyModel, AssetOccupancy)> OccupyAsync(OccupyAssetInfoModel model,
         Guid? occupierUserId)
     {
-        // Todo: lock the day?
         var asset = await AssetRepository.GetAsync(model.AssetId);
         var category = await AssetCategoryRepository.GetAsync(asset.AssetCategoryId);
         var occupyingModel = new ProviderOccupyingInfoModel(asset, category, model.StartingTime, model.Duration,
             model.Date, model.Volume);
-        var result = await InternalCanOccupyAsync(occupyingModel);
-        await HandleCanOccupyResultAsync(result);
-        return await InternalOccupyAsync(occupyingModel, occupierUserId);
+
+        var periodScheme = await GetEffectivePeriodSchemeAsync(model.Date, asset, category);
+        var handle = await DistributedLock.TryAcquireAsync(
```

Comment:
I think this lock is meaningless. Users try to occupy, and they succeed or fail. How does the lock affect it?

Reply:
I agree that the `DbConcurrencyException` is unfriendly. Is it necessary to be in the `AssetOccupancyProviderBase`? It's only a problem of the default asset occupancy provider

Reply:
@gdlcf88 I have add UnitTest to cover this scenario.

https://github.com/EasyAbp/BookingService/blob/7e00d006ac2d75b793c8efbf11ad5cc7c5c4bfa1/test/EasyAbp.BookingService.Domain.Tests/AssetOccupancyProviders/DefaultAssetOccupancyProviderTests/OccupyTests.cs#L187